README.md

```diff
@@ -124,6 +125,41 @@ Key:
 * `=` Maintenance is manual, only severe security bugs will be patched.
 * `-` Deprecated; please upgrade.
 
+#### Dependency Matrix
+Since there are multiple libraries that client-go is either required to, or can work
+with, this is a list of all the libraries and versions that work with each other.
+
+| client-go    | [apimachinery](https://github.com/kubernetes/apimachinery) | [apiextensions-apiserver](https://github.com/kubernetes/apiextensions-apiserver)<sup>*</sup> | [code-generator](https://github.com/kubernetes/code-generator)<sup>+</sup> |
+| ----------------------------------| -----------------------------------------------------------|----------------------------------------------------------------------------------------------|----------------------------------------------------------------------------|
+| tag: v5.0.x  | tag: kubernetes-1.8.x  | tag: kubernetes-1.8.x | tag: kubernetes-1.8.x |
+| tag: v6.0.x  | tag: kubernetes-1.9.x  | tag: kubernetes-1.9.x | tag: kubernetes-1.9.x |
+
+<sup>*</sup> optional library that exposes the Extensions API, for example, for 
+Custom Resource Definitions.  
+<sup>+</sup> optional library for generating Kubernetes style API types. Typically 
+used with Custom Resource Definitions.
+
+Note that `x` versions need to align. For example tag v.5.0.1 in client-go, will be
+supported with tag kubernetes-1.8.1 in apimachinery and other repositories
+
+Using the git tag versions is the preferred way of managing dependencies for newer 
+releases of client-go, and since client-go is backward compatible, it is recommended that
+developers always use the latest version.
```

Comment:
This paragraph doesn't seem to be necessary? The compatibility matrix states what version of client-go a user needs.